src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/service/MessageNotifier.kt

```diff
@@ -34,10 +48,51 @@ class MessageNotifier(
         .stringValue(hearingEventType.description)
         .build()
 
+    when {
+      messageLargerThanThreshold(hearingEvent) -> publishLargeMessage(hearingEventType = hearingEventType, hearingEvent = hearingEvent)
+      else -> publishMessage(hearingEvent, messageTypeValue, hearingEventTypeValue)
+    }
+  }
+
+  private fun publishMessage(hearingEvent: HearingEvent, messageTypeValue: MessageAttributeValue, hearingEventTypeValue: MessageAttributeValue) {
     val publishResult = topic.publish(eventType = "commonplatform.case.received", event = objectMapper.writeValueAsString(hearingEvent), attributes = mapOf("messageType" to messageTypeValue, "hearingEventType" to hearingEventTypeValue), messageGroupId = MESSAGE_GROUP_ID)
+
     log.info("Published message with message Id {}", publishResult.messageId())
   }
 
+  fun publishLargeMessage(hearingEventType: HearingEventType, hearingEvent: HearingEvent) {
+    val hearingEventTypeValue =
```

Comment:
see line 45 - why not just pass this in as a messageAttribute?
Also do we not need to set messageType as we do on the small messages?